packages/stream-metadata/src/routes/spaceImage.ts

```diff
@@ -19,30 +18,68 @@ const paramsSchema = z.object({
 	eventId: z.string().optional(),
 })
 
+const querySchema = z.object({
+	size: z.string().optional(),
+})
+
 const CACHE_CONTROL = {
-	// Client caches for 30s, uses cached version for up to 7 days while revalidating in background
-	307: 'public, max-age=30, s-maxage=3600, stale-while-revalidate=604800',
+	// Original size - longer cache since immutable
+	200: 'public, max-age=31536000, immutable',
+	// Resized images - shorter cache but still long-term for CDN
+	'200-resized': 'public, max-age=86400, s-maxage=31536000',
 	400: 'public, max-age=30, s-maxage=3600',
 	404: 'public, max-age=5, s-maxage=3600', // 5s max-age to avoid client's rendering broken images during town creation flow
 	422: 'public, max-age=30, s-maxage=3600',
+	500: 'public, max-age=30, s-maxage=3600',
 }
 
 export async function fetchSpaceImage(request: FastifyRequest, reply: FastifyReply) {
 	const logger = request.log.child({ name: fetchSpaceImage.name })
 
-	const parseResult = paramsSchema.safeParse(request.params)
+	const paramsResult = paramsSchema.safeParse(request.params)
+	const queryResult = querySchema.safeParse(request.query)
 
-	if (!parseResult.success) {
-		const errorMessage = parseResult.error.errors[0]?.message || 'Invalid parameters'
+	if (!paramsResult.success) {
+		const errorMessage = paramsResult.error.errors[0]?.message || 'Invalid parameters'
 		logger.info(errorMessage)
 		return reply
 			.code(400)
 			.header('Cache-Control', CACHE_CONTROL[400])
 			.send({ error: 'Bad Request', message: errorMessage })
 	}
 
-	const { spaceAddress, eventId } = parseResult.data
-	logger.info({ spaceAddress, eventId }, 'Fetching space image')
+	if (!queryResult.success) {
```

Comment:
do we require size params now? seems like we should keep this backwards compatible